rts/lra/lra-client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -230,6 +226,27 @@ public void setCurrentLRA(URI coordinatorUri) {
         }
     }
 
+    public List<LRAData> getAllLRAs() {
+        Client client = null;
+        try {
+            client = ClientBuilder.newClient();
+            Response response = client.target(base)
+                .request()
+                .get();
+
+            if (response.getStatus() != Response.Status.OK.getStatusCode()) {
+                LRALogger.i18NLogger.error_getAllLRAs(response.getStatus());
+                throw new WebApplicationException(response);
+            }
+
+            return response.readEntity(new GenericType<List<LRAData>>() {});
+        } finally {
+            if (client != null) {
+                client.close();
```

Comment:
@mmusgrov sorry, but I would prefer to keep this for now and really investigate the TCK hangs with Client caching in new JBTM. Would that be ok with you?

Reply:
You did half of what I asked (getClient()) but not the other half. You either need both changes or neither. The reason I asked to get the client via a method is so that we can cache it in the future. The finally block needs to call a method so that any future strategy for caching the connection will be in one place.

Reply:
Which hangs? I am talking about opening and closing clients using a single method. The reason is as follows:

The docs (https://docs.oracle.com/javaee/7/api/javax/ws/rs/client/package-summary.html) say "... Client instances - configurable, heavy-weight objects that manage the underlying communication infrastructure ...".

That indicates to me that we need to be configuring a fixed minimal set of clients and we then share them thus avoiding the set up and tear down costs. If we put the relevant logic behind a method call then we can more easily decide how we want to do that. The original code (that this PR is reinstating) was already doing that (see pr/1515).

Reply:
The change I am requesting is a structural change as opposed to a functional one so I am resolving this conversation.